big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= O(n)
+    i++                             // 2 * n = O(2n) ~= O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n) ~= O(n)
+    if (!seen[word]) {              // 2 * n = O(2n) ~= O(n)
+      seen[word] = true;            // 2 * 1 = O(2) ~= O(1)
+      unique[unique.length] = word; // 3 * n = O(3n) ~= O(n)
+    }
+  }
+  return unique;                    // 1 * 1 = O(1)
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= O(n)
+    i++                             // 2 * n = O(2n) ~= O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n) ~= O(n)
+    let isUnique = true;            // 1 * n = O(n)
+    for (
+      let c = 0;                    // 1 * n = O(n)
+      c < i;                        // 2 * n * n = O(2n^2) ~= O(n^2)
+      c++                           // 2 * n * n = O(2n^2) ~= O(n^2)
+    ) {
+      const comparing = words[c];   // 2 * n * n = O(2n^2) ~= O(n^2)
+      if (comparing === word) {     // 3 * n * n = O(3n^2) ~= O(n^2)
+        isUnique = false;           // 1 * 1 = O(1)
```

Comment:
This one, also.
When doing complexity analysis, you should assume that every branch is executed every time through the loop. At the end of the day, it doesn't make a difference, though.